diffsync/store/redis.py

```diff
@@ -0,0 +1,242 @@
+"""RedisStore module."""
+import copy
+import uuid
+from pickle import loads, dumps  # nosec
```

Comment:
Since DiffSyncModels are Pydantic classes, it shouldn't be necessary to use `pickle` if security is a concern - you should be able to instead use `model.json()` to dump to JSON, and `<model_class>.parse_raw()` to instantiate it from JSON. Not sure how comparable the performance will be versus pickle, but worth considering?

Reply:
I was not aware of the `parse_raw()` helper method in pydantic models. Good one.
However, I have a question. If I use the `model.json()` method I'm losing some fields vs `dict()`, that are not simple to convert to json:

```
(Pdb) obj.json()
'{"name": "site1"}'
(Pdb) obj.dict()
{'model_flags': <DiffSyncModelFlags.NONE: 0>, 'name': 'site1', 'devices': []}
```

This was the reason to use pickle, and get the raw objects directly. Do you see a proper way to overcome that?

Reply:
That's surprising to me that `json()` is discarding properties from `dict()`, I wouldn't have expected that from reading the Pydantic docs. I'm fine with sticking with `pickle` if the baseline json/dict serialization is not adequate to the task.

Reply:
I guess that the values that can't be serialized, are discarded.